analyses/cell-type-ewings/exploratory_analysis/04-infercnv.Rmd

```diff
@@ -0,0 +1,485 @@
+The rest of the analysis will use the results from running `InferCNV` with endothelial cells as the reference. 
+
+## Using CNVs to annotate tumor cells 
+
+`InferCNV` uses an HMM-based model to predict whether or not a copy number variation (either loss or gain) is present for each chromosome in each cell. 
+For each predicted CNV region, a posterior probability is calculated for that CNV for each cell.
+
+From the [`InferCNV` documentation](https://github.com/broadinstitute/inferCNV/wiki/inferCNV-HMM-based-CNV-Prediction-Methods): 
+
+> CNV regions with mean posterior probabilities of being normal (no CNV) that are above a maximum threshold are removed as likely false positive predictions.
+
+Before we can make any plots we need to add some metadata that contains CNV information on a per cell level to the cell metadata.
+
+```{r}
+# make df for plotting
+coldata_df <- sce |> 
+  scuttle::makePerCellDF(use.dimred = "UMAP") |> 
+  # replace UMAP.1 with UMAP1
+  dplyr::rename_with(
+        \(x) stringr::str_replace(x, "^UMAP\\.", "UMAP")
+      ) |> 
+  # add in per cell cnv information 
+  dplyr::left_join(cnv_metadata_df, by = c("barcodes"))
+
+# remove MT and GL chr, comments in docs recommend that these are not useful and can be inaccurate 
+chr_columns_to_keep <- which(!stringr::str_detect(names(coldata_df), "chrMT|chrGL000219.1|chrGL000194.1"))
+all_cnv_df <- coldata_df[, chr_columns_to_keep]
+```
+
+The metadata that we added includes the following columns for each chromosome: 
+
+- `has_cnv_chr1`
+- `has_loss_chr1`
+- `has_dupli_chr1`
+- `proportion_cnv_chr1`
+- `proportion_loss_chr1`
+- `proportion_dupli_chr1`
+- `proportion_scaled_cnv_chr1`
+- `proportion_scaled_loss_chr1`
+- `proportion_scaled_dupli_chr1`
+
+The `has` columns are boolean columns indicating if that cell has presence of a CNV, loss, or duplication event. 
+If `has_cnv` is TRUE if either `has_loss` or `has_dupli` is TRUE. 
+
+See [the comments in this issue](https://github.com/broadinstitute/infercnv/issues/208) explaining what the contents of the columns are. 
+
+One approach to classifying tumor cells is to look for cells that have a lot of CNVs. 
+Let's start by just summing up the total number of chromosomes with CNVs per cell and plotting that. 
+
+
+```{r}
+# grab any columns that have has_cnv 
+has_cnv_cols <- names(all_cnv_df)[startsWith(colnames(all_cnv_df), "has_cnv")]
+
+# don't include the normal cells that were provided as reference in scaling 
+reference_cells_only <- all_cnv_df |> 
+  dplyr::filter(stringr::str_detect(subcluster, "reference")) |> 
+  dplyr::mutate(has_cnv = NA,
+                z_scaled_cnv = NA)
+
+unknown_cells_only <- all_cnv_df |>
+  dplyr::filter(stringr::str_detect(subcluster, "unknown")) |> 
+  # get total number of chromosomes with cnvs per cell and then scale that number 
+  dplyr::mutate(has_cnv = rowSums(across(has_cnv_cols)),
+                z_scaled_cnv = scale(has_cnv))
```

Comment:
Rather than splitting these up and then rejoining, I would probably do something like this:
```suggestion
# don't include the normal cells that were provided as reference in scaling 
cnv_df <- all_cnv_df |>
  # get total number of chromosomes with cnvs per cell and then scale that number 
  dplyr::mutate(
    has_cnv = ifelse(
      stringr::str_detect(subcluster, "reference"), 
      NA_real_, 
      rowSums(across(has_cnv_cols))
    ),
    z_scaled_cnv = scale(has_cnv))
```

`scale` unlike `mean` and `sum` seems to handle `NA` values as you would want it to.